state/block-executor.md

```diff
@@ -0,0 +1,52 @@
+# Block Executor
+
+## Abstract
+
+The `BlockExecutor` is a component responsible for creating, applying, and maintaining blocks and state in the system. It interacts with the mempool and the application via the [ABCI interface].
+
+## Component Description
+
+The `BlockExecutor` is initialized with a proposer address, `namespace ID`, `chain ID`, `mempool`, `proxyApp`, `eventBus`, and `logger`. It uses these to manage the creation and application of blocks. It also validates blocks and commits them, updating the state as necessary.
+
+## Detailed Description
```

Comment:
Can we make this one section instead? The `Component Description` looks v similar to the `Abstract` right now.